data_safe_haven/config/config_sections.py

```diff
@@ -59,3 +60,11 @@ class ConfigSectionSRE(BaseModel, validate_assignment=True):
     software_packages: SoftwarePackageCategory = SoftwarePackageCategory.NONE
     timezone: TimeZone = "Etc/UTC"
     workspace_skus: list[AzureVmSku] = Field(..., default_factory=list[AzureVmSku])
+
+    @fieldvalidator("admin_ip_addresses", "data_provider_ip_addresses", "research_user_ip_addresses", mode="after")
+    @classmethod
+    def ensure_non_overlapping(cls, v: list[IpAddress]) -> list[IpAddress]:
```

Comment:
This doesn't account for overlaps like `1.1.0.0/24` and `1.1.0.20/32`. This probably needs the functions in `IPv4Network` from `ipaddress`.

Reply:
They are strings, but they're CIDRs. You will need to convert them to `IPv4Network`s then check for overlap.